src/types/widget.ts

```diff
@@ -270,7 +270,12 @@ export type Widget<
 /**
  * Transforms the given items.
  */
-export type TransformItems<TItem> = (items: TItem[]) => TItem[];
+export type TransformItems<TItem, TResults = SearchResults<TItem>> = (
+  items: TItem[],
+  metadata: {
+    results: TResults | undefined;
+  }
+) => TItem[];
```

Comment:
```suggestion
export type TransformItems<TItem, TMetaData = {
    results?: SearchResults;
  }> = (
  items: TItem[],
  metadata: TMetadata
) => TItem[];
```

we shouldn't link item to results, as item is only the hits inside the hits/infinitehits component, where you can pass the generic explicitly by overriding TMetadata. What do you think?